demo/fake_server.ex

```diff
@@ -56,22 +56,97 @@ defmodule Wanda.Executions.FakeServer do
 
   defp complete_fake_execution(execution_id, group_id, targets) do
     result = Enum.random([:passing, :warning, :critical])
-    check_results = build(:check_results_from_targets, targets: targets, result: result)
+    check_results = check_results_from_targets(targets, result)
 
     build_result =
       build(:result,
         check_results: check_results,
         execution_id: execution_id,
-        group_id: group_id,
-        result: result
+        group_id: group_id
       )
 
     Executions.complete_execution!(
       execution_id,
-      build_result
+      Map.put(build_result, :result, get_result(build_result))
     )
 
     execution_completed = Messaging.Mapper.to_execution_completed(build_result)
     :ok = Messaging.publish("results", execution_completed)
   end
+
+  defp get_result(%Result{check_results: check_results}) do
+    check_results
+    |> Enum.map(& &1.result)
+    |> Enum.map(&{&1, result_weight(&1)})
+    |> Enum.max_by(fn {_, weight} -> weight end)
+    |> elem(0)
+  end
+
+  defp result_weight(:critical), do: 2
+  defp result_weight(:warning), do: 1
+  defp result_weight(:passing), do: 0
+
+  defp check_results_from_targets(targets, result) do
+    targets
+    |> Enum.flat_map(& &1.checks)
+    |> Enum.uniq()
+    |> Enum.map(fn check_id ->
+      check_targets = Enum.filter(targets, &(check_id in &1.checks))
+      check_result_from_target(check_id, check_targets, result)
+    end)
+  end
+
+  defp expectation_evaluations_from_check(check, result) do
+    Enum.map(check.expectations, fn expectation ->
+      case expectation.type do
+        :expect ->
+          build(:expectation_evaluation,
+            type: :expect,
+            name: expectation.name,
+            return_value: result == :passing
+          )
+
+        :expect_same ->
+          build(:expectation_evaluation,
+            type: :expect_same,
+            name: expectation.name,
+            return_value:
```

Comment:
What for is this `if/esle` with the hardcoded the `same_value`?
As far as I see, the value will be the same in all cases, as the the same expectation_evaluation is attached to all targets.
Maybe we could simply put the faked value as `return_value`.